test/fixtures/dialects/ansi/create_table_constraint_reference_option.sql

```diff
@@ -0,0 +1,4 @@
+CREATE TABLE b (
+    b INT NOT NULL,
+    CONSTRAINT `c_a` FOREIGN KEY (`b`) REFERENCES `a`(`b`) ON DELETE CASCADE ON UPDATE SET NULL
+)
```

Comment:
If you make the other suggested changes in this review and use ANSI syntax (below) this will parse fine
```sql
CREATE TABLE test_table (
    test_col INT NOT NULL,
    CONSTRAINT test_contstraint FOREIGN KEY (test_col) REFERENCES other_table(other_column) ON DELETE CASCADE ON UPDATE SET NULL
)
```